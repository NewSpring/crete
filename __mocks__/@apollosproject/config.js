const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    SHARE_URL: 'https://newspring.cc',
    TIMEZONE: 'America/New_York',
  },
  ROCK_CONSTANTS: {
    IMAGE: 10,
    AUDIO_FILE: 77,
    VIDEO_FILE: 79,
  },
  ROCK_MAPPINGS: {
    FEED_CONTENT_CHANNEL_IDS: [1, 2, 3, 4, 6, 8],
    SERIES_CONTENT_CHANNEL_TYPE_IDS: [6, 7],
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [18, 20],
      },
      DevotionalContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [17],
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
      WeekendContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
});

export default ApolloServer;
