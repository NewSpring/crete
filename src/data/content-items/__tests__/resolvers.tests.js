import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';

import { createGlobalId } from '@apollosproject/server-core';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';

import {
  featuresSchema,
  mediaSchema,
  themeSchema,
  scriptureSchema,
  liveSchema,
  peopleSchema,
} from '@apollosproject/data-schema';

import { ContentChannel, Sharable } from '@apollosproject/data-connector-rock';

import * as ContentItem from '../index';

class Cache {
  get = () => Promise.resolve(null);

  set = () => Promise.resolve(null);

  initialize({ context }) {
    this.context = context;
  }
}

const { getSchema, getContext } = createTestHelpers({
  ContentChannel,
  ContentItem,
  Sharable,
  Cache: { dataSource: Cache },
});

const contentItemFragment = `
  fragment ContentItemFragment on ContentItem {
    id
    __typename
    title
    summary
    coverImage {
      name
      key
      sources {
        uri
      }
    }
    images {
      __typename # Typenames here to increase test coverage
      name
      key
      sources {
        __typename
        uri
      }
    }
    videos {
      __typename
      name
      key
      sources {
        __typename
        uri
      }
      embedHtml
    }
    audios {
      __typename
      name
      key
      sources {
        __typename
        uri
      }
    }
    htmlContent
    childContentItemsConnection {
      edges {
        node {
          id
          __typename
        }
        cursor
      }
      pageInfo {
        startCursor
        endCursor
      }
    }
    parentChannel {
      id
      __typename
    }
    sharing {
      __typename
      url
      title
      message
    }
  }
`;

describe('UniversalContentItem', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([
      featuresSchema,
      themeSchema,
      mediaSchema,
      scriptureSchema,
      liveSchema,
      peopleSchema,
    ]);
    context = getContext();
    context.dataSources.ContentItem.getShareUrl = jest.fn(
      () => 'https://newspring.cc/whatever'
    );
    context.dataSources.ContentItem.getCommunicators = jest.fn(() => [
      {
        firstName: 'first',
        lastName: 'last',
      },
    ]);
    context.dataSources.ContentItem.getGuestCommunicators = jest.fn(() => [
      'guest communicator',
    ]);
    context.dataSources.ContentItem.getContentItemScriptures = jest.fn(() => [
      {
        html: '<p><i>1</i>In the beginning...</p>',
      },
    ]);
  });

  it('gets a newspring content item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
          ...ContentItemFragment
        }
      }
      ${contentItemFragment}
    `;

    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a newspring MediaContentItem item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'MediaContentItem')}") {
          ...ContentItemFragment
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a newspring ContentSeriesContentItem item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'ContentSeriesContentItem')}") {
          ...ContentItemFragment
        }
      }
      ${contentItemFragment}
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a newspring devotional item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(123, 'DevotionalContentItem')}") {
          id
          ... on DevotionalContentItem {
            id
            title
            scriptures {
              html
            }
          }
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a newspring weekend content item', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'WeekendContentItem')}") {
          id
          ... on WeekendContentItem {
            title
            communicators {
              firstName
              lastName
            }
            guestCommunicators
            sermonDate
            features {
              __typename
              id
              ... on ScriptureFeature {
                scriptures {
                  reference
                }
              }
              ... on TextFeature {
                body
              }
            }
          }
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    console.log('result = ', result);
    expect(result).toMatchSnapshot();
  });
});
