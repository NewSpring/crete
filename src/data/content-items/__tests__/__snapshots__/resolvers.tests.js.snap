// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversalContentItem gets a newspring ContentSeriesContentItem item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "ContentSeriesContentItem",
      "audios": Array [],
      "childContentItemsConnection": Object {
        "edges": Array [
          Object {
            "cursor": "b487224762b030f470967f45d7205823",
            "node": Object {
              "__typename": "UniversalContentItem",
              "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "b487224762b030f470967f45d7205823",
          "startCursor": "b487224762b030f470967f45d7205823",
        },
      },
      "coverImage": null,
      "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
      "id": "ContentSeriesContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "images": Array [],
      "parentChannel": Object {
        "__typename": "ContentChannel",
        "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
      },
      "sharing": Object {
        "__typename": "SharableContentItem",
        "message": "SAMPLE: Easter - Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "title": "Share via ...",
        "url": "https://newspring.cc/whatever",
      },
      "summary": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "title": "SAMPLE: Easter",
      "videos": Array [],
    },
  },
}
`;

exports[`UniversalContentItem gets a newspring MediaContentItem item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "MediaContentItem",
      "audios": Array [],
      "childContentItemsConnection": Object {
        "edges": Array [
          Object {
            "cursor": "b487224762b030f470967f45d7205823",
            "node": Object {
              "__typename": "UniversalContentItem",
              "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "b487224762b030f470967f45d7205823",
          "startCursor": "b487224762b030f470967f45d7205823",
        },
      },
      "coverImage": null,
      "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
      "id": "MediaContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "images": Array [],
      "parentChannel": Object {
        "__typename": "ContentChannel",
        "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
      },
      "sharing": Object {
        "__typename": "SharableContentItem",
        "message": "SAMPLE: Easter - Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "title": "Share via ...",
        "url": "https://newspring.cc/whatever",
      },
      "summary": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "title": "SAMPLE: Easter",
      "videos": Array [],
    },
  },
}
`;

exports[`UniversalContentItem gets a newspring content item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "UniversalContentItem",
      "audios": Array [],
      "childContentItemsConnection": Object {
        "edges": Array [
          Object {
            "cursor": "b487224762b030f470967f45d7205823",
            "node": Object {
              "__typename": "UniversalContentItem",
              "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "b487224762b030f470967f45d7205823",
          "startCursor": "b487224762b030f470967f45d7205823",
        },
      },
      "coverImage": null,
      "htmlContent": "<p>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sollicitudin condimentum aliquet. In est nulla, lacinia ac dictum et, laoreet vitae elit. Proin tempus tellus ligula, a consequat diam consectetur a. Phasellus luctus velit sed lorem mollis commodo. Nunc sit amet blandit velit. Donec tincidunt congue facilisis. Sed iaculis at neque non porttitor. Phasellus ultrices egestas erat feugiat pellentesque. Duis venenatis, dolor quis fringilla tempus, sem lorem euismod lectus, sed egestas felis magna at felis. Pellentesque ut rhoncus erat, a pulvinar purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Ut sit amet consequat est. Maecenas et porta dui, non condimentum lectus.</p>
<p>
	Suspendisse vel nibh odio. Pellentesque porta sapien ligula, in laoreet diam tempus sed. Morbi nunc erat, mattis eu pulvinar blandit, adipiscing quis magna. Ut quis dui lobortis velit suscipit consectetur. Nulla iaculis fermentum egestas. Aenean venenatis sagittis mauris, sed rhoncus purus accumsan ac. Suspendisse potenti. Sed sed tempor turpis. Duis sit amet nisi nec purus fringilla condimentum. Phasellus non lacus arcu. Donec scelerisque, erat sed tempor elementum, nulla risus scelerisque ante, ac imperdiet velit magna ut quam. Nam tristique orci auctor consequat laoreet. Quisque malesuada metus sed sodales eleifend. Aenean rhoncus, mi sit amet ullamcorper tincidunt, sem sem rutrum felis, in semper enim massa ut sem.</p>
<p>
	Vivamus diam urna, cursus in sapien in, porta gravida enim. Cras non fringilla arcu, tincidunt laoreet lacus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Aliquam volutpat felis quis augue faucibus ultrices. Morbi lobortis vestibulum sodales. Sed tincidunt urna vitae felis ultrices, pharetra placerat quam dignissim. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel adipiscing tellus. In vitae sodales diam. Aliquam pharetra orci a porta molestie. In et neque bibendum, viverra leo sit amet, auctor magna. Morbi posuere massa sed metus euismod, et adipiscing sem dictum. Cras eget elementum risus, non imperdiet ligula.</p>
",
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "images": Array [],
      "parentChannel": Object {
        "__typename": "ContentChannel",
        "id": "ContentChannel:744e1011cfa6a90bb2fbfeb3ecdc1451",
      },
      "sharing": Object {
        "__typename": "SharableContentItem",
        "message": "SAMPLE: Easter - Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "title": "Share via ...",
        "url": "https://newspring.cc/whatever",
      },
      "summary": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "title": "SAMPLE: Easter",
      "videos": Array [],
    },
  },
}
`;

exports[`UniversalContentItem gets a newspring devotional item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "DevotionalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "scriptures": Array [
        Object {
          "html": "<p><i>1</i>In the beginning...</p>",
        },
      ],
      "title": "SAMPLE: Easter",
    },
  },
}
`;

exports[`UniversalContentItem gets a newspring weekend content item 1`] = `
Object {
  "data": Object {
    "node": Object {
      "communicators": Array [
        Object {
          "firstName": "first",
          "lastName": "last",
        },
      ],
      "features": null,
      "guestCommunicators": Array [
        "guest communicator",
      ],
      "id": "WeekendContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "sermonDate": null,
      "title": "SAMPLE: Easter",
    },
  },
}
`;
